### PR TITLE
Async Crypto Endeavour

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
Got all tests passing locally with a small fix to ipfs-repo which I'm creating a PR for now

**Comment by dignifiedquire** [comment]
PR is up: https://github.com/ipfs/js-ipfs-repo/pull/102

**Comment by diasdavid** [comment]
This is the last giant to defeat in order to finish the Async Crypto PR Mountain :)

File: src/decision/engine.js [hunk 1]
```diff
@@ -26,37 +29,45 @@ module.exports = class Engine {
...
         }
 
-        const nextTask = this.peerRequestQueue.pop()
+        nextTask = this.peerRequestQueue.pop()
+        log('check', this._running && nextTask)
+        return Boolean(nextTask)
+      },
+      (next) => {
+        log('generating')
         log('got task', nextTask)
```
Comment (diasdavid) [inline src/decision/engine.js]: two logs next to each other, better coalesce to one

Comment (dignifiedquire) [inline src/decision/engine.js]: ahhhh, there I don't wait for you one single time..

Comment (diasdavid) [inline src/decision/engine.js]: :)

Comment (dignifiedquire) [inline src/decision/engine.js]: fixed in master

**Pull request merged**